test/integration/buildruns_to_taskruns_test.go

```diff
@@ -117,13 +121,25 @@ var _ = Describe("Integration tests BuildRuns and TaskRuns", func() {
 						buildRunWitcher.Stop()
 					}
 				}
-
-				Expect(len(seq)).To(Equal(2))
-				Expect(seq[0].Type).To(Equal(v1alpha1.Succeeded))
-				Expect(seq[0].Status).To(Equal(corev1.ConditionUnknown))
-				Expect(seq[0].Reason).To(Equal("Pending"))
-				Expect(seq[1].Type).To(Equal(v1alpha1.Succeeded))
-				Expect(seq[1].Reason).To(Equal("Running"))
+				// consider a longer sequence, for events where the cluster have
+				// insufficient resources, where the Reason will be ExceededNodeResources.
+				Expect(len(seq) >= 2).To(Equal(true))
+
+				// ensure the conditions move eventually from unknown into running
+				Expect(seq).Should(ContainElement(&v1alpha1.Condition{
+					Type:               v1alpha1.Succeeded,
+					Status:             corev1.ConditionUnknown,
+					LastTransitionTime: metav1.Time{Time: fakeTime},
+					Reason:             "Pending",
```

Comment:
I think this Reason might also be ExceededNodeResources in some cases (see #570). Can this test just check for `Status:Unknown` and ignore any other differences? Tekton might also change exactly the Reasons and Messages it reports, so being flexible with those changes might be better.

Reply:
The original idea of the test was to ensure we have a transition of the `Reason` field under the same `Status:Unknown`. What we observed was that a TaskRun always moves from `Reason:Pending` to `Reason:Running`. Only asserting for the `Status:Unknown` will not reflect our initial assumption. Therefore a Tekton change in `Reason` will break this test and will notify us of it. What do you think?

Reply:
How much does it matter to Shipwright's users what the underlying TaskRun Reason is? I see that the BuildRun controller is just copying the TaskRun Reason and Message directly over to the BuildRun, but I think long-term you might want to move away from that, for two reasons:

1. Tekton might add more state transitions, that BuildRun users don't care about. Today it's `Reason:Pending` -> `Reason:Running`, but in the future it might be `Reason:Pending` -> `Reason:Scheduling` -> `Reason:Optimizing` -> `Reason: PettingKittens` -> `Reason:Running` -- BuildRuns might not care about all of those reasons, they just want to know if it's working or not -- in particular, to know whether logs might be available.
1. BuildRun might want to add more Reasons that don't cleanly map to TaskRun Reasons. BuildRun might want to add a `Reason: DetectingStrategy` or something.

I don't think it's necessary to make this separation _right now_, but long-term I think it could be useful. Maybe BuildRuns surface TaskRun `Reason: ExceededNodeResources` as BuildRun `Reason: Pending` as a first step, and this e2e test just tests that the BuildRun goes `Pending` -> `Running`, whatever the underlying Tekton machinery is doing.

This abstraction of the lower layers gives you flexibility to adapt to changes in those lower layers, including even possibly not depending on Tekton someday (please don't though! 😄). This also gives Tekton more flexibility to change how it reports statuses, without fear of causing churn for Shipwright or its dependents.

Sorry this reply got so long, let me know what you think. This change definitely lgtm, especially if it results in fewer flakes.

Reply:
@ImJasonH thanks, I couldn't agree more with what you are saying. I think decoupling our conditions from the TaskRun ones is a good idea. I open an issue to provide a follow-up on your feedback https://github.com/shipwright-io/build/issues/575 .

I would like to get this PR merged atm, to avoid hitting this flaky tests in the future.